runtime/src/bank/pyth_accumulator_tests.rs

```diff
@@ -0,0 +1,904 @@
+            emitter_chain: 26,
+            emitter_address: ACCUMULATOR_EMITTER_ADDRESS,
+            payload: expected_accumulator.serialize(bank.slot(), ACCUMULATOR_RING_SIZE),
+            ..Default::default()
+        },
+    };
+
+    assert_eq!(
+        updated_wormhole_message_account.data(),
+        expected_wormhole_message.try_to_vec().unwrap()
+    );
+}
+
+#[test]
+fn test_accumulator_v2_all_v2() {
+    test_accumulator_v2([false, false, false, false]);
+}
+
+#[test]
+fn test_accumulator_v2_all_v1() {
+    test_accumulator_v2([true, true, true, true]);
+}
+
+#[test]
+fn test_accumulator_v2_mixed() {
+    test_accumulator_v2([true, true, false, false]);
+}
+
+fn test_accumulator_v2(generate_buffers: [bool; 4]) {
+    let leader_pubkey = solana_sdk::pubkey::new_rand();
+    let GenesisConfigInfo {
+        mut genesis_config, ..
+    } = create_genesis_config_with_leader(5, &leader_pubkey, 3);
+
+    // Set epoch length to 32 so we can advance epochs quickly. We also skip past slot 0 here
+    // due to slot 0 having special handling.
+    let slots_in_epoch = 32;
+    genesis_config.epoch_schedule = EpochSchedule::new(slots_in_epoch);
+    let mut bank = Bank::new_for_tests(&genesis_config);
+
+    let generate_price = |seeds, generate_buffers: bool| {
+        let (price_feed_key, _bump) = Pubkey::find_program_address(&[seeds], &ORACLE_PID);
+        let mut price_feed_account =
+            AccountSharedData::new(42, size_of::<PriceAccount>(), &ORACLE_PID);
+
+        let messages = {
+            let price_feed_info_key = &price_feed_key.to_bytes().into();
+            let price_feed_info_lamports = &mut 0;
+            let price_feed_info_owner = &ORACLE_PID.to_bytes().into();
+            let price_feed_info_data = price_feed_account.data_mut();
+            let price_feed_info = AccountInfo::new(
+                price_feed_info_key,
+                false,
+                true,
+                price_feed_info_lamports,
+                price_feed_info_data,
+                price_feed_info_owner,
+                false,
+                Epoch::default(),
+            );
+
+            let mut price_account = PriceAccount::initialize(&price_feed_info, 0).unwrap();
+            if !generate_buffers {
+                price_account.flags.insert(
+                    PriceAccountFlags::ACCUMULATOR_V2 | PriceAccountFlags::MESSAGE_BUFFER_CLEARED,
+                );
+            }
+
+            vec![
+                price_account
+                    .as_price_feed_message(&price_feed_key.to_bytes().into())
+                    .to_bytes(),
+                price_account
+                    .as_twap_message(&price_feed_key.to_bytes().into())
+                    .to_bytes(),
+            ]
+        };
+
+        bank.store_account(&price_feed_key, &price_feed_account);
+
+        if generate_buffers {
+            // Insert into message buffer in reverse order to test that accumulator
+            // sorts first.
```

Comment:
where are you reversing it :?

Reply:
This was left over after copy-paste. Removed.